### PR TITLE
Wffhcohort 458

### PR DESCRIPTION
**Pull request by shelleyMurphy**
We can't get rid of InputStreams entirely, and in the end I opted to leave a lot of the methods that allow translation given an InputStream etc, but I replaced any case where you might access the sources on a provider with a String instead.

### CODE REVIEW

File: cohort-engine-api-web/src/test/resources/cql/dependency_test-1.0.0/dependency_test-1.0.0.cql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+library "Dependency Test" version '1.0.0'
```
Comment (csandersdev) [inline cohort-engine-api-web/src/test/resources/cql/dependency_test-1.0.0/dependency_test-1.0.0.cql:1]: This library name doesn't match up casing-wise with the filename. Is the discrepancy intentional? Looks like you might be testing an edge case, which would certainly be a good thing, but it caught my eye as incorrect.

File: cohort-engine/src/main/java/com/ibm/cohort/engine/MultiFormatLibrarySourceProvider.java [hunk 2]
```diff
@@ -73,35 +78,44 @@ public InputStream getLibrarySource(VersionedIdentifier libraryIdentifier, Libra
...
+			specFormat.put(LibraryFormat.XML, fhirHelperResourceAsString);
 			InputStream fhirHelperResourceCQL = MultiFormatLibrarySourceProvider.class.getResourceAsStream(
 					String.format("/org/hl7/fhir/%s-%s.cql",
 							libraryIdentifier.getId(),
 							libraryIdentifier.getVersion()));
-			specFormat.put(LibraryFormat.CQL, fhirHelperResourceCQL);
+			String fhirHelperResourceCQLAsString = new BufferedReader(
+					new InputStreamReader(fhirHelperResourceCQL, StandardCharsets.UTF_8))
+					.lines()
+					.collect(Collectors.joining("\n"));
```
Comment (csandersdev) [inline cohort-engine/src/main/java/com/ibm/cohort/engine/MultiFormatLibrarySourceProvider.java:117]:
```suggestion
					.collect(Collectors.joining(System.lineSeparator()));
```

File: cohort-engine/src/main/java/com/ibm/cohort/engine/MultiFormatLibrarySourceProvider.java [hunk 3]
```diff
@@ -73,35 +78,44 @@ public InputStream getLibrarySource(VersionedIdentifier libraryIdentifier, Libra
...
 		if(specFormat.isEmpty()) {
 			InputStream fhirHelperResource = MultiFormatLibrarySourceProvider.class.getResourceAsStream(
 					String.format("/org/hl7/fhir/%s-%s.xml",
 							libraryIdentifier.getId(),
 							libraryIdentifier.getVersion()));
-			specFormat.put(LibraryFormat.XML, fhirHelperResource);
+			String fhirHelperResourceAsString = new BufferedReader(
+					new InputStreamReader(fhirHelperResource, StandardCharsets.UTF_8))
+					.lines()
+					.collect(Collectors.joining("\n"));
```
Comment (csandersdev) [inline cohort-engine/src/main/java/com/ibm/cohort/engine/MultiFormatLibrarySourceProvider.java:108]:
```suggestion
					.collect(Collectors.joining(System.lineSeparator()));
```

File: cohort-engine/src/test/java/com/ibm/cohort/engine/TestClasspathLibrarySourceProvider.java [hunk 4]
```diff
@@ -17,12 +21,16 @@ public TestClasspathLibrarySourceProvider(List<String> libraryResources,
...
+			Map<LibraryFormat, String> formats = sources.computeIfAbsent(vid, key -> new HashMap<>());
 			InputStream is = ClassLoader.getSystemResourceAsStream(resource);
 			if( is == null ) {
 				throw new IllegalArgumentException( resource );
 			}
-			formats.put(LibraryFormat.forString(resource), is);
+			String resourceAsString = new BufferedReader(
+					new InputStreamReader(is, StandardCharsets.UTF_8))
+					.lines()
+					.collect(Collectors.joining("\n"));
```
Comment (csandersdev) [inline cohort-engine/src/test/java/com/ibm/cohort/engine/TestClasspathLibrarySourceProvider.java:32]:
```suggestion
					.collect(Collectors.joining(System.lineSeparator()));
```

File: cohort-engine-api-web/src/test/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandlerTest.java [hunk 5]
```diff
@@ -758,6 +758,80 @@ public void testCohortEvaluation() throws Exception {
 		ServiceBaseUtility.apiCleanup(Mockito.any(), Mockito.eq(MethodNames.EVALUATE_COHORT.getName()));
 	}
 
+	@PrepareForTest({ Response.class, FHIRRestUtils.class })
+	@Test
+	public void testCohortMultipleDependencies() throws Exception {
```
Comment (csandersdev) [inline cohort-engine-api-web/src/test/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandlerTest.java:763]: Was this test added to account for the multiple read issue? Just want to make sure we have coverage for the future.

File: cohort-engine-api-web/src/test/resources/cql/dependency_test-1.0.0/dependency_test-1.0.0.cql (hunk 1 above)
Comment (shelleyMurphy) [reply cohort-engine-api-web/src/test/resources/cql/dependency_test-1.0.0/dependency_test-1.0.0.cql:1]: No, since I'm mocking returns it didn't matter but let me change it for consistency's sake

File: cohort-engine-api-web/src/test/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandlerTest.java (hunk 5 above)
Comment (shelleyMurphy) [reply cohort-engine-api-web/src/test/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandlerTest.java:763]: Yeah, this test makes sure the input stream isn't already "spent", and in master it fails.

File: cohort-engine/src/main/java/com/ibm/cohort/engine/MultiFormatLibrarySourceProvider.java (hunk 3 above)
Comment (shelleyMurphy) [reply cohort-engine/src/main/java/com/ibm/cohort/engine/MultiFormatLibrarySourceProvider.java:108]: your suggestions have been committed, thanks!

File: cohort-engine/src/main/java/com/ibm/cohort/engine/MultiFormatLibrarySourceProvider.java [hunk 6]
```diff
@@ -37,10 +42,10 @@
 	 * @param sourceFormat      Source format of the stream of interest
 	 * @return InputStream where source data can be read.
```
Comment (msargentibm) [inline cohort-engine/src/main/java/com/ibm/cohort/engine/MultiFormatLibrarySourceProvider.java:43]: I noticed some of the javadocs still call out `InputStream` where the return type has changed.

Comment (shelleyMurphy) [reply cohort-engine/src/main/java/com/ibm/cohort/engine/MultiFormatLibrarySourceProvider.java]: Whoops. Fixed!

**Review by msargentibm — APPROVED** [review]
